src/types/contract.rs

```diff
@@ -1,42 +1,160 @@
+pub use build_info::BuildInfo;
 use serde::{Deserialize, Serialize};
 
 /// The struct provides information about deployed contract's source code and supported standards.
 ///
-/// Contract source metadata follows [NEP-330 standard](https://github.com/near/NEPs/blob/master/neps/nep-0330.md) for smart contract verification
+/// Contract source metadata follows [**NEP-330 standard**](https://github.com/near/NEPs/blob/master/neps/nep-0330.md) for smart contracts
+///
+/// See documentation of [`crate::Contract::contract_source_metadata`] on how to query this for a contract via this crate
+// `rustdoc` clearly lacks functionality of automatic backlinks within a single crate
 #[derive(Debug, Clone, PartialEq, Default, Eq, Serialize, Deserialize)]
 pub struct ContractSourceMetadata {
-    /// Optional version identifier, typically a commit hash or semantic version
+    /// Optional version identifier, typically a semantic version
+    ///
+    /// ## Examples:
     ///
-    /// Examples:
-    /// - Git commit: "39f2d2646f2f60e18ab53337501370dc02a5661c"
-    /// - Semantic version: "1.0.0"
+    /// ```rust,no_run
+    /// # let version: Option<String> =
+    /// // Semantic version
+    /// Some("1.0.0".into())
+    /// # ;
+    /// ```
     pub version: Option<String>,
 
     // cSpell::ignore bafybeiemxf5abjwjbikoz4mc3a3dla6ual3jsgpdr4cjr3oz3evfyavhwq
-    /// Optional URL to source code repository or IPFS CID
+    /// Optional URL to source code repository/tree
+    ///
+    /// ## Examples:
     ///
-    /// Examples:
-    /// - GitHub URL: "<https://github.com/near-examples/nft-tutorial>"
-    /// - IPFS CID: "bafybeiemxf5abjwjbikoz4mc3a3dla6ual3jsgpdr4cjr3oz3evfyavhwq"
```

Comment:
ipfs cid doesn't really qualify well to be a `link` thing, as it has no protocol identificator it's somehow related to ipfs, and there's no context to derive this knowledge from.

returned the example nonetheless